platform/default/online_file_source.cpp

```diff
@@ -332,9 +332,7 @@ void OnlineFileSource::Impl::reschedule(OnlineFileRequestImpl& request) {
 
     const Seconds timeout = request.getRetryTimeout();
 
-    if (timeout == Seconds::zero()) {
-        update(request);
```

Comment:
It's used by `OnlineFileSource::Impl::add`. After I merge this I'll inline it there.